articles/openstack/ostack-how-cloud-config.md

```diff
@@ -0,0 +1,66 @@
+---
+title: How to configure an OpenStack instance using cloud-config | UKCloud Ltd
+description: Describes how to use cloud-config (user-data) to configure an OpenStack instance
+services: openstack
+author: Steve Relf
+toc_rootlink: How To
+toc_sub1:
+toc_sub2:
+toc_sub3:
+toc_sub4:
+toc_title: Configure an instance using cloud-config
+toc_fullpath: How To/ostack-how-cloud-config.md
+toc_mdlink: ostack-how-cloud-config.md
+---
+
+# How to configure an OpenStack instance using cloud-config
+
+## Overview
+
+When you boot an instance in OpenStack you can pass configuration to the instance as cloud-config (sometimes called user-data). This enables you to quickly and easily make configuration changes at boot time.
+
+## Use cases
+
+The following list provides some examples of how you can use cloud-config to configure an instance:
+
+- Setting a default locale
+- Setting an instance hostname
+- Generating instance SSH private keys
+- Adding SSH keys to a user's `.ssh/authorized_keys` so they can log in
+- Setting up ephemeral mount points
+- Configuring network devices
+- Running config scripts at startup
+
```

Comment:
Unfortunately, I've been asked about cloud-config scripts 4 times; 3 of those times were about how to set passwords in instances.

I'm guessing we're consciously not mentioning that as a use case since it's a bad idea?